hail/src/main/scala/is/hail/expr/ir/EmitStream.scala

```diff
@@ -731,7 +731,7 @@ object EmitStream {
     def step(idx: S)(k: Step[A, S] => Code[Ctrl])(implicit ctx: EmitStreamContext): Code[Ctrl] = {
       val eos = ctx.jb.joinPoint()
       val yld = ctx.jb.joinPoint[A](ctx.mb)
-      eos.define { _ => k(EOS) }
+      eos.define { _ => Code(yld.init, k(EOS)) }
```

Comment:
@patrick-schultz I ran into a bytecode verification error for `ArrayZip(Seq(MakeStream(...), MakeStream(...)), ...)` because statically it can appear like one or more of the streams hit EOS before you evaluate the body.  I know you've fixed this in the new setup.  This is the best fix I could come up with, but isn't great (JoinPoint.init just calls the parameter pack init).  Do yo have a better way to do this?

Reply:
Initializing the locals of a JoinPoint seems very odd to me. The locals are always defined at the top of the JoinPoint body, and only used inside the body, so I don't understand where the bytecode verification error could come from. I'll see if I can replicate this and understand it.

On the other hand, this code doesn't have long to live, so if this works, it's probably fine.

Reply:
The code is a array zip of two streams.  It looks something like:

```
Lstream1:
  ... goto Lyeild1 or Leos1
Lyield1:
  ... write parameter pack
  allEOS = 0
  goto Lstream2
Leos1:
  ... don't write parameter pack
  anyEOS = 1
  goto Lstream2
# stream 2 the same, goes to Lzip after yield/eos
Lzip:
  if allEOS:
    goto LzipEOS
  else if anyEOS:
    fatal
  else:
    goto LzipYield
```

So it looks like there is a code path from Lstream1 to LzipYield that goes through Leos for the zip input streams.  The parameterpack never gets written if Yield is never called.

Reply:
I want to put the init of the parameter pack in the init of the stream, but the code organization doesn't make that possible (I can't construct the join point outside the step function to use in the init function).

Reply:
Also, this default value initialization stuff is obviously a crap constraint to put on the code generator and the Code  should just deal with it.

Reply:
Yeah, I see the problem. The new framework does make this much easier to handle. There's probably a more "proper" way to fix it here, but I don't think it's worth spending time to find.